thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/DetectionUtils.java

```diff
@@ -288,4 +288,39 @@ public static Period periodFromTimeUnit(int size, TimeUnit unit) {
         return new Period(TimeUnit.MILLISECONDS.convert(size, unit));
     }
   }
+
+  /**
+   * Aggregate the time series data frame's value to specified granularity
+   * @param df the data frame
+   * @param origin the aggregation origin time stamp
+   * @param granularityPeriod the aggregation granularity in period
+   * @return the aggregated time series data frame
+   */
+  public static DataFrame aggregateByPeriod(DataFrame df, long origin, Period granularityPeriod) {
+    return df.groupByPeriod(df.getLongs(COL_TIME), new DateTime(origin), granularityPeriod).sum(COL_TIME, COL_VALUE);
```

Comment:
Should we check the aggregation function? Are we assuming the aggregation is always SUM?

Reply:
Good catch. Maybe we need to support AVG too?

Reply:
Can we pass the aggregation function as an argument

Reply:
Sure. Changed to that and also support AVG